DoublyLinkedList.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * DoublyLinkedList.c
+ *
+ *  Created on: Nov 3, 2022
+ *      Author: Mohamed Refat
+ */
+
+#include <stdlib.h>
+#include "STD_TYPES.h"
+#include "DoublyLinkedList.h"
+#include <assert.h>
+void DoublyLinkedList_Init(DoublyLinkedList_t *pList)
+{
+	pList->pHead = NULL_ptr;
+	pList->pTail = NULL_ptr;
+	pList->Size  = 0;
+}
+
+uint8_t DoublyLinkedList_InsertNode(DoublyLinkedList_t *pList , uint32_t Position , LINKEDLIST_TYPE Data )
+{
+	Node_t *pLocNode ;
+	Node_t *pLocTempNodeNode ;
+	uint8_t LocReturn;
+
+
+	if( (pLocNode = (Node_t *)malloc(sizeof(Node_t))) )
```

Comment:
Your implementation is totally right but it can be tricky for another developer or for you after long time as you used to see logical operator  `==`  inside `if` condition so by default you (or another developer) will change it and your code will fail😢 
To git rid of this complex situation you have to make sth like that : 
```
  pLocNode = (Node_t *)malloc(sizeof(Node_t);
  if( pLocNode == 1) {
```